src/main/java/org/candlepin/subscriptions/retention/TallyRetentionPolicy.java

```diff
@@ -103,4 +103,8 @@ public OffsetDateTime getCutoffDate(Granularity granularity) {
             String.format("Unsupported granularity: %s", granularity));
     }
   }
+
+  public long getSnapshotsToDeleteInBatches() {
```

Comment:
Can we apply some validation to this setting?  I know we do validation in stuff like `ConduitFacts`, but I can't remember if we do it (yet) on any of the configuration settings.  Regardless, we definitely want to make sure this value is greater than zero somehow.  I prefer bean validation annotations, but if that's a pain then just a check in the setter and an exception thrown at startup is fine.  A typo like '000000' would result in an infinite loop.

Reply:
Good point! Added.